scalding-parquet/src/main/scala/com/twitter/scalding/parquet/tuple/scheme/ParquetTupleConverter.scala

```diff
@@ -0,0 +1,89 @@
+package com.twitter.scalding.parquet.tuple.scheme
+
+import parquet.io.api.{ Binary, Converter, GroupConverter, PrimitiveConverter }
+
+import scala.collection.mutable
+import scala.util.Try
+
+/**
+ * Parquet tuple converter used to create user defined tuple value from parquet column values
+ * @param parent parent parquet tuple converter
+ * @param isOption is the field optional
+ * @param outerIndex field index in parent tuple schema
+ */
+abstract class ParquetTupleConverter(val parent: Option[ParquetTupleConverter] = None, val isOption: Boolean = false,
+  val outerIndex: Int = -1) extends GroupConverter {
+  var converters: Map[Int, Converter] = Map()
+  val fieldValues: mutable.ArrayBuffer[Any] = mutable.ArrayBuffer()
```

Comment:
The downside of this is boxing primitives.
temporarily storing the value before creating the case class instance could be delegated to the PrimitiveConverter children. Each can have a field of the right type. There's no need to store it as an Option as we will know to wrap it at the time of creation of the case class instance. ```Option(value)```